src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.DateTimeException;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Represents a Storage object that handles file IO for tasks.
+ */
+public class Storage {
+    private final Path path;
+
+    public Storage() {
+        this.path = Paths.get(".", "db.txt");
+    }
+
+    public Storage(Path path) {
+        this.path = path;
+    }
+
+    /**
+     * Saves given list of tasks to file specified at instantiation.
+     * If file does not exist, it will be created.
+     * The list of tasks will be formatted in easily parsable format delimited
+     * by |.
+     *
+     * @param taskList The list of tasks to be saved.
+     * @throws IOException If something goes wrong with File operations.
+     */
+    public void save(List<Task> taskList) throws IOException {
+        boolean fileExists = Files.exists(path);
```

Comment:
Would it better to rename fileExists to `hasFile` instead? This would adhere to the coding standard of using is, has, was at the start of boolean names.